sighupio/fury-eks-installer
# feat(eks-logs): add variable to enable or disable logs flags

This pull request updates our EKS Terraform configurations to enhance flexibility for logs configuration.
**Actions taken:**
- Introduced a new variable `cluster_enabled_log_types` for dynamic log type management.
- Implemented a validation rule for the new variable to ensure only supported log types are used.
- Updated EKS example configurations to utilize the new variable, replacing hardcoded log types.
- Applied `terraform fmt --recursive` to ensure consistent code formatting.

For more details on the log types supported by Amazon EKS, please refer to the [official documentation](https://docs.aws.amazon.com/eks/latest/userguide/control-plane-logs.html).

**Variable Documentation:**
```hcl
variable "cluster_enabled_log_types" {
  description = "Specifies which Kubernetes control plane logs to enable. Supports 'api', 'audit', 'authenticator', 'controllerManager', and 'scheduler'. Can be left empty to disable logging."
  type        = list(string)
  default     = ["api", "audit", "authenticator", "controllerManager", "scheduler"]

  validation {
    condition     = length(var.cluster_enabled_log_types) == 0 || alltrue([for log_type in var.cluster_enabled_log_types: contains(["api", "audit", "authenticator", "controllerManager", "scheduler"], log_type)])
    error_message = "Invalid log type. Must be one of 'api', 'audit', 'authenticator', 'controllerManager', 'scheduler', or an empty list."
  }
}
```
**QA Testing:**

The validation of the `cluster_enabled_log_types` variable was thoroughly tested to ensure its correctness. The tests were conducted using the Terraform version specified in the repository's `.tool-versions` file. It's important to note that since this update pertains to an existing configuration within the module, there was no need to recreate an entire cluster. Instead, the focus was on enhancing the way the logging flag is defined to make it more dynamic and flexible.

Specifically, the tests covered the following scenarios:
- Validation of an empty list to ensure logging is disabled when desired.
- Verification that the variable correctly accepts and processes valid log types individually and in combination.
- Confirmation that the validation rule effectively blocks invalid log type entries and provides the expected error message.

These tests ensured that the validation of the `cluster_enabled_log_types` variable functions as intended, guaranteeing accurate and reliable log type configuration.
